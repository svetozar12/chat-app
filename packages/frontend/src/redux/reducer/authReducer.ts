import { Action } from "../state";
import { ActionType } from "../state";
import { InitialState } from "../state";
import { HYDRATE } from "next-redux-wrapper";

const initialState = {
  remember_me: false,
  input: "",
  good: "", //good and bad stand for good alert and bad alert
  bad: "",
  loginPrompt: false,
  cookie: "",
};
const reducer = (state: InitialState = initialState, action: Action) => {
  switch (action.type) {
    case HYDRATE:
      return {
        ...state,
        ...action.payload,
      };
    case ActionType.LOGIN_POST:
      return { ...state, state };
    case ActionType.LOGIN_POST_ERROR:
      return { ...state, bad: action.bad };
    case ActionType.REGISTER_POST:
      return { ...state, good: action.good };
    case ActionType.REGISTER_POST_ERROR:
      return { ...state, bad: action.bad };
    case ActionType.SAVE_INPUT:
      return { ...state, input: action.payload };
    case ActionType.REMEMBER_ME_CHECK:
      return { ...state, remember_me: action.payload };
    case ActionType.QUICK_LOGIN:
      return { ...state, loginPrompt: action.payload };
    case ActionType.SIGN_IN:
      return { ...state, cookie: action.payload };
    case ActionType.SIGN_OUT:
      return { state: undefined };
    default:
      return state;
  }
};

export default reducer;
