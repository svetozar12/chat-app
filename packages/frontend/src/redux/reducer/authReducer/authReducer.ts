import { HYDRATE } from "next-redux-wrapper";
import { IAuthState } from "./state";
import { ActionType } from "../../types";
export const initialState: IAuthState = {
  remember_me: false,
  good: "", //good and bad stand for good alert and bad alert
  bad: "",
  loginPrompt: false,
  cookie: {
    username: "",
    id: "",
    token: "",
    refresh_token: "",
  },
};
const reducer = (state = initialState, action: any) => {
  switch (action.type) {
    case HYDRATE:
      return {
        ...state,
        ...action.payload,
      };
    case ActionType.LOGIN_POST:
      return { ...state, state };
    case ActionType.LOGIN_POST_ERROR:
      return { ...state, bad: action.bad };
    case ActionType.REGISTER_POST:
      return { ...state, good: action.good };
    case ActionType.REGISTER_POST_ERROR:
      return { ...state, bad: action.bad };
    case ActionType.REMEMBER_ME_CHECK:
      return { ...state, remember_me: action.payload };
    case ActionType.QUICK_LOGIN:
      return { ...state, loginPrompt: action.payload };
    case ActionType.SIGN_IN:
      return { ...state, cookie: action.payload };
    default:
      return state;
  }
};

export default reducer;
export { initialState as AuthState };
