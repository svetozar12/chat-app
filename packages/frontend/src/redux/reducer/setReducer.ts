import { Action, ActionType, InitialState2 } from "../state";
export const initialState = {
  reciever: "",
  chat_inviter: "",
  pageNumber: 2,
  setUserSettings: false,
  setFriendRequest: false,
  toggleCreateGroup: false,
  setMobileNav: false,
  setIsMatch: false,
  setChatSettings: false,
  setModalInvite: false,
  setIsLoggingIn: false,
};
const setReducer = (state: InitialState2 | any | InitialState2[] = initialState, action: Action) => {
  switch (action.type) {
    case ActionType.SET_RECIEVER:
      return { ...state, reciever: action.payload };
    case ActionType.CHAT_INVITER:
      return { ...state, chat_inviter: action.payload };
    case ActionType.INCREMENT_PAGE_NUMBER:
      return { ...state, pageNumber: ++action.payload };
    case ActionType.SET_FRIEND_REQUEST:
      return { ...state, setFriendRequest: action.payload };
    case ActionType.SET_USER_SETTINGS:
      return { ...state, setUserSettings: action.payload };
    case ActionType.TOGGLE_CREATE_GROUP:
      return { ...state, toggleCreateGroup: action.payload };
    case ActionType.SET_MOBILE_NAV:
      return { ...state, setMobileNav: action.payload };
    case ActionType.SET_CHAT_SETTINGS:
      return { ...state, setChatSettings: action.payload };
    case ActionType.SET_IS_MATCH:
      return { ...state, setIsMatch: action.payload };
    case ActionType.SET_MODAL_INVITE:
      return { ...state, setModalInvite: action.payload };
    case ActionType.SET_IS_LOGEDIN:
      return { ...state, setIsLoggingIn: action.payload };
    case ActionType.SIGN_OUT:
      return initialState;
    default:
      return state;
  }
};

export default setReducer;
