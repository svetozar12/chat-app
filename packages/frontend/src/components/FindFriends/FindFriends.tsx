import React from "react";
import { Socket } from "socket.io-client";
import { useDispatch, useSelector } from "react-redux";
import { IInitialSet } from "../../redux/reducer/setReducer/state";
import ISave_inputState from "../../redux/reducer/save_inputReducer/state";
import { BsSearch, BsThreeDots } from "react-icons/bs";
import { AiOutlineUsergroupAdd } from "react-icons/ai";
import { IoNotifications } from "react-icons/io5";
import { FaUserCircle } from "react-icons/fa";
import { css, cx } from "@emotion/css";
import UserSettings from "../UserSettings";
import { IAuthState } from "../../redux/reducer/authReducer/state";
import api_helper from "../../graphql/api_helper";

export interface IFindFriends {
  cookieName: string;
  socketRef: Socket;
  cookie: any;
}

function FindFriends({ socketRef }: IFindFriends) {
  const dispatch = useDispatch();
  const [hasAvatar, setHasAvatar] = React.useState(false);
  const [image, setImage] = React.useState("");
  const authState = useSelector((state: { authReducer: IAuthState }) => state.authReducer);

  const getUserImage = async () => {
    try {
      const res = await api_helper.user.getById(authState.cookie?.id as string, authState.cookie?.token as string);
      const userAvatar = res.data.user.userAvatar;
      if (!userAvatar) {
        setHasAvatar(false);
        return true;
      }
      setHasAvatar(true);
      const requestString = ``;
      setImage(requestString);
      return true;
    } catch (error) {
      return false;
    }
  };

  React.useEffect(() => {
    getUserImage();
  }, []);

  const state = useSelector((state: { setReducer: IInitialSet }) => state.setReducer);

  const notifState = useSelector((state: { saveInputReducer: ISave_inputState }) => state.saveInputReducer);

  const toggleGroupCreate = () => {
    dispatch({
      type: "TOGGLE_CREATE_GROUP",
      payload: !state.toggleCreateGroup,
    });
    dispatch({
      type: "SET_MOBILE_NAV",
      payload: !state.setMobileNav,
    });
    dispatch({
      type: "SET_CHAT_SETTINGS",
      payload: false,
    });
  };

  const sendInvite = async () => {
    try {
      const res = await api_helper.invite.create(authState.cookie?.id as string, state.reciever, authState.cookie?.token as string);
      const data = res.data.message;
      socketRef.emit("send_friend_request", {
        inviter: data.inviter,
        reciever: state.reciever,
      });

      return true;
    } catch (error) {
      return false;
    }
  };
  const handleSubmit = async (e: React.FormEvent<HTMLFormElement> | React.MouseEvent<SVGElement>) => {
    e.preventDefault();
    if (state.reciever) {
      await sendInvite();
      dispatch({ type: "SET_RECIEVER", payload: "" });
    }
  };

  return (
    <form
      className={css`
        width: 100%;
        margin: 0;
        padding: 1rem;
        display: flex;
        justify: flex-start;
        position: relative;
        margin-bottom: 1rem;
        align-items: center;
      `}
      onSubmit={handleSubmit}
    >
      <div
        className={css`
          width: 95%;
          height: 3rem;
          margin-bottom: 1rem;
          position: relative;
        `}
      ></div>
      <div
        className={css`
          display: flex;
          width: 95%;
          justify-content: space-between;
          padding-bottom: 1rem;
          border-bottom: 1px solid rgba(0, 0, 0, 0.1);
          margin: 0 1rem;
        `}
      >
        {/* refactor to new component bellow */}
        <div className="flex">
          {hasAvatar ? (
            <img
              alt="user_avatar"
              src={image}
              className={cx(
                "click",
                css`
                  width: 2.8rem;
                  height: 2.8rem;
                  color: var(--main-logo-color);
                  margin-right: 1rem;
                  border-radius: 100px;
                  position: relative;
                  zindex: 10;
                  &:hover {
                    background: rgba(122, 122, 122, 0.4);
                  }
                `,
              )}
            />
          ) : (
            <FaUserCircle
              className={css`
                width: 3.25rem;
                height: 3.25rem;
                margin: 0;
                color: var(--main-logo-color);
                position: relative;
                z-index: 10;
              `}
            />
          )}
          <h1 style={{ whiteSpace: "nowrap", margin: "0 0 0 1rem " }}>Chats</h1>
        </div>
        {/* refactor to new component above */}
        <div className="flex find_friends_icons">
          <div
            className="flex notifications"
            style={{ cursor: "pointer" }}
            onClick={() => {
              dispatch({
                type: "SET_FRIEND_REQUEST",
                payload: !state.setFriendRequest,
              });
            }}
          >
            <IoNotifications />
            {notifState.notification_number != 0 && <div className="flex">{notifState.notification_number}</div>}
          </div>
          <div
            className="flex add_group"
            style={{
              cursor: "pointer",
            }}
            onClick={toggleGroupCreate}
          >
            <AiOutlineUsergroupAdd />
          </div>
          <div
            className="flex dots"
            style={{
              cursor: "pointer",
              position: "relative",
            }}
          >
            <BsThreeDots
              onClick={() =>
                dispatch({
                  type: "SET_USER_SETTINGS",
                  payload: !state.setUserSettings,
                })
              }
            />
            {state.setUserSettings ? <UserSettings /> : null}
          </div>
        </div>
      </div>
      <div className="flex" style={{ width: "95%", position: "relative" }}>
        <div
          className={css`
            width: 95%;
            border-radius: 25px;
            margin: 1rem 0;
            padding: 0.3rem;
            background: rgba(122, 122, 122, 0.1);
            display: flex;
            justify-content: center;
            align-items: center;
            &:focus-within {
              box-shadow: 0 0 5px;
            }
          `}
        >
          <BsSearch style={{ cursor: "pointer", color: "black" }} onClick={handleSubmit} />
          <input
            className={css`
              width: 90%;
              margin: 0 1rem;
              background: transparent;
              border: none;
              outline: none;
            `}
            onChange={(e) => dispatch({ type: "SET_RECIEVER", payload: e.target.value })}
            placeholder="Search for chats"
            value={state.reciever}
            type="search"
          />
        </div>
      </div>
    </form>
  );
}

export default FindFriends;
