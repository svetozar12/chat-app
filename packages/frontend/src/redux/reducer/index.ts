import { combineReducers } from "redux";
import reducer from "./authReducer";
import setReducer from "./setReducer";
import saveInputReducer from "./save_inputReducer";
import messageReducer from "./messageReducer";

const reducers = combineReducers({
  authReducer: reducer,
  setReducer,
  messageReducer,
  saveInputReducer,
});

export default reducers;
