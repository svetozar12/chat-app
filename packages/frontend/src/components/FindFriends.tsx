import React from "react";
import axios from "axios";
import { Socket } from "socket.io-client";
import { useDispatch, useSelector } from "react-redux";
import { InitialState2, InitialState3 } from "../redux/state";
import { requestUrl } from "../utils/hostUrl_requestUrl";
import { BsFillPeopleFill, BsSearch } from "react-icons/bs";
import { IoNotifications } from "react-icons/io5";
import { FaUserCircle, FaUserFriends } from "react-icons/fa";
import UserSettings from "./UserSettings";

export interface IFindFriends {
  cookieName: string;
  socketRef: Socket;
  cookie: any;
}

function FindFriends({ cookie, cookieName, socketRef }: IFindFriends) {
  const dispatch = useDispatch();
  const [hasAvatar, setHasAvatar] = React.useState(false);
  const [image, setImage] = React.useState("");

  const getUserImage = async (name: string) => {
    try {
      const res = await axios.get(`${requestUrl}/users/${name}`);
      const userAvatar = res.data.user.userAvatar;
      if (!userAvatar) {
        setHasAvatar(false);
        return true;
      }
      setHasAvatar(true);
      const requestString = `${requestUrl}/${userAvatar}`;
      setImage(requestString);
      return true;
    } catch (error) {
      return false;
    }
  };

  React.useEffect(() => {
    getUserImage(cookieName);
  }, []);

  const state = useSelector(
    (state: { setReducer: InitialState2 }) => state.setReducer,
  );

  const notifState = useSelector(
    (state: { saveInputReducer: InitialState3 }) => state.saveInputReducer,
  );

  const toggleGroupCreate = () => {
    dispatch({
      type: "TOGGLE_CREATE_GROUP",
      payload: !state.toggleCreateGroup,
    });
    dispatch({
      type: "SET_MOBILE_NAV",
      payload: !state.setMobileNav,
    });
    dispatch({
      type: "SET_CHAT_SETTINGS",
      payload: false,
    });
  };

  const sendInvite = async () => {
    try {
      const res = await axios.post(`${requestUrl}/invites`, {
        reciever: state.reciever,
        inviter: cookieName,
        status: "recieved",
      });
      const data = res.data.message;
      socketRef.emit("send_friend_request", {
        inviter: data.inviter,
        reciever: state.reciever,
      });

      return true;
    } catch (error) {
      return false;
    }
  };
  const handleSubmit = async (
    e: React.FormEvent<HTMLFormElement> | React.MouseEvent<SVGElement>,
  ) => {
    e.preventDefault();
    if (state.reciever) {
      await sendInvite();
      dispatch({ type: "SET_RECIEVER", payload: "" });
    }
  };

  return (
    <form
      className="friendsInput"
      onSubmit={handleSubmit}
      style={{ marginBottom: "1rem", alignItems: "center" }}
    >
      <div className="close_hamburger">
        <h1 style={{ whiteSpace: "nowrap" }}>Your chats</h1>
      </div>
      <div className="profile" style={{ display: "flex" }}>
        <div>
          {hasAvatar ? (
            <img
              src={image}
              onClick={() =>
                dispatch({
                  type: "SET_USER_SETTINGS",
                  payload: !state.setUserSettings,
                })
              }
              className="user-logo click"
              style={{
                width: "3rem",
                height: "3.25rem",
                borderRadius: "100px",
                cursor: "pointer",
                position: "relative",
                zIndex: "10",
              }}
            />
          ) : (
            <FaUserCircle
              style={{
                color: "var(--main-logo-color)",
                position: "relative",
                zIndex: "10",
                width: "3rem",
                height: "3.25rem",
              }}
              onClick={() =>
                dispatch({
                  type: "SET_USER_SETTINGS",
                  payload: !state.setUserSettings,
                })
              }
            />
          )}
          {state.setUserSettings ? <UserSettings cookie={cookie} /> : null}
        </div>
        <div className="flex find_friends_icons">
          <div
            className="notifications"
            style={{ cursor: "pointer" }}
            onClick={() => {
              dispatch({
                type: "SET_FRIEND_REQUEST",
                payload: !state.setFriendRequest,
              });
            }}
          >
            <IoNotifications />
            {notifState.notification_number != 0 && (
              <div className="flex">{notifState.notification_number}</div>
            )}
          </div>
          <div
            style={{
              cursor: "pointer",
            }}
            onClick={toggleGroupCreate}
          >
            <BsFillPeopleFill />
          </div>
        </div>
      </div>
      <div className="flex search_container">
        <div className="search-bar">
          <BsSearch
            style={{ cursor: "pointer", color: "black" }}
            onClick={handleSubmit}
          />
          <input
            onChange={(e) =>
              dispatch({ type: "SET_RECIEVER", payload: e.target.value })
            }
            value={state.reciever}
            type="search"
          />
        </div>
      </div>
    </form>
  );
}

export default FindFriends;
