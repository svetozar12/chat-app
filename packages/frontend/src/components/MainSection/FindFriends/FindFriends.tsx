import React from 'react';
import { connect } from 'react-redux';
import { css } from '@emotion/css';
import { bindActionCreators, Dispatch } from 'redux';
import { Skeleton, VStack, FormControl } from '@chakra-ui/react';
import { useCookie } from 'next-cookie';
// services
import { setReciever } from 'services/redux/reducer/invites/actions';
import { getAuth } from 'utils/authMethods';
import { useAuth } from 'utils/SessionProvider';
import { STATE } from 'services/redux/reducer';
import { IWebSocket } from 'services/redux/reducer/websocket/state';
import IInvite from 'services/redux/reducer/invites/state';
import sdk from 'services/sdk';
// components
import SkeletonFindFriendsHeader from 'components/Loading/SkeletonFindFriendsHeader';
import FindFriendsHeader from './FindFriendsHeader';
import FindFriendsSearch from './FindFriendsSearch';

interface IFindFriends {
  ws: IWebSocket;
  invite: IInvite;
  setReciever: typeof setReciever;
}

function FindFriends(props: IFindFriends) {
  const { ws, invite, setReciever } = props;

  const cookie = useCookie();
  const { user } = useAuth();

  const sendInvite = async () => {
    try {
      await getAuth();
      const res = await sdk.invite.create({ userId: cookie.get('id'), AccessToken: cookie.get('token') }, invite.reciever);

      const { reciever, inviter } = res;

      ws.ws?.emit('send_friend_request', {
        inviter,
        reciever,
      });

      return true;
    } catch (error) {
      console.log(error);

      return false;
    }
  };
  const handleSubmit = async (e: React.FormEvent<HTMLFormElement> | React.MouseEvent<SVGElement>) => {
    e.preventDefault();
    if (invite.reciever) {
      await sendInvite();
      setReciever('');
    }
  };

  return (
    <VStack
      className={css`
        width: 100%;
        margin: 0;
        padding: 1rem;
        display: flex;
        justify: flex-start;
        position: relative;
        margin-bottom: 1rem;
        align-items: center;
      `}
    >
      {user ? <FindFriendsHeader /> : <SkeletonFindFriendsHeader />}
      <FormControl>{user ? <FindFriendsSearch handleSubmit={handleSubmit} /> : <Skeleton w="100%" mt={1} h="2.6875rem" />}</FormControl>
    </VStack>
  );
}

const mapStateToProps = (state: STATE) => ({
  ws: state.ws,
  invite: state.invite,
});

const mapDispatchToProps = (dispatch: Dispatch) => ({
  setReciever: bindActionCreators(setReciever, dispatch),
});

export default connect(mapStateToProps, mapDispatchToProps)(React.memo(FindFriends));
