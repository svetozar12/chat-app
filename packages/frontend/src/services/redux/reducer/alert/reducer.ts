import { AnyAction } from 'redux';
import { IAlert } from 'services/redux/reducer/alert/state';
import { ActionType } from '../../types';

const initialState: IAlert = {
  good: '', // good and bad stand for good alert and bad alert
  bad: '',
};

const alertReducer = (state = initialState, action: AnyAction) => {
  switch (action.type) {
    case ActionType.LOGIN_POST_ERROR:
      return { ...state, bad: action.payload };
    case ActionType.REGISTER_POST:
      return { ...state, good: action.payload };
    case ActionType.REGISTER_POST_ERROR:
      return { ...state, bad: action.payload };
    default:
      return state;
  }
};

export default alertReducer;
