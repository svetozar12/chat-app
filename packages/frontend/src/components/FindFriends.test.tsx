import configureStore from "redux-mock-store";
import FindFriends from "../../src/components/FindFriends";
import renderer from "react-test-renderer";
import { Provider } from "react-redux";
import { initialState as setState } from "../../src/redux/reducer/setReducer";
import { AuthState } from "../../src/redux/reducer/authReducer";
import saveInputReducer from "../../src/redux/reducer/save_inputReducer";
import { screen, render, cleanup, RenderResult } from "@testing-library/react";
import { ReactTestRendererJSON } from "react-test-renderer";
import "@testing-library/jest-dom";

let component: ReactTestRendererJSON | ReactTestRendererJSON[] | null;
let container: RenderResult;
const socketRef: any = jest.fn();
const cookie: any = jest.fn();
beforeEach(() => {
  const mockStore = configureStore([]);
  const store = mockStore({
    authReducer: AuthState,
    setReducer: setState,
    saveInputReducer: saveInputReducer,
  });

  component = renderer
    .create(
      <Provider store={store}>
        <FindFriends cookie={cookie} cookieName="ivan" socketRef={socketRef} />
      </Provider>,
    )
    .toJSON();

  container = render(
    <Provider store={store}>
      <FindFriends cookie={cookie} cookieName="ivan" socketRef={socketRef} />
    </Provider>,
  );
});

afterEach(cleanup);

describe("Render connected React-redux page", () => {
  it("should create snapshot for <FindFriends/>", () => {
    expect(component).toMatchSnapshot();
  });

  it("should render <FindFriends/>", () => {
    const renderedComponent = container.getByRole("searchbox");
    expect(renderedComponent).toBeInTheDocument();
  });
});
