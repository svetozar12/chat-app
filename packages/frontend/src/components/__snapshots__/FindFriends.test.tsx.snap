// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render connected React-redux page should create snapshot for <FindFriends/> 1`] = `
<form
  className="css-1om3h1z"
  onSubmit={[Function]}
  style={
    Object {
      "alignItems": "center",
      "marginBottom": "1rem",
    }
  }
>
  <div
    className="css-1q40041"
  >
    <h1
      style={
        Object {
          "whiteSpace": "nowrap",
        }
      }
    >
      Your chats
    </h1>
  </div>
  <div
    className="css-1j8uik1"
  >
    <div>
      <svg
        className="css-1hbbq6s"
        fill="currentColor"
        height="1em"
        onClick={[Function]}
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 496 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M248 8C111 8 0 119 0 256s111 248 248 248 248-111 248-248S385 8 248 8zm0 96c48.6 0 88 39.4 88 88s-39.4 88-88 88-88-39.4-88-88 39.4-88 88-88zm0 344c-58.7 0-111.3-26.6-146.5-68.2 18.8-35.4 55.6-59.8 98.5-59.8 2.4 0 4.8.4 7.1 1.1 13 4.2 26.6 6.9 40.9 6.9 14.3 0 28-2.7 40.9-6.9 2.3-.7 4.7-1.1 7.1-1.1 42.9 0 79.7 24.4 98.5 59.8C359.3 421.4 306.7 448 248 448z"
        />
      </svg>
    </div>
    <div
      className="flex find_friends_icons"
    >
      <div
        className="notifications"
        onClick={[Function]}
        style={
          Object {
            "cursor": "pointer",
          }
        }
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 512 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M440.08 341.31c-1.66-2-3.29-4-4.89-5.93-22-26.61-35.31-42.67-35.31-118 0-39-9.33-71-27.72-95-13.56-17.73-31.89-31.18-56.05-41.12a3 3 0 01-.82-.67C306.6 51.49 282.82 32 256 32s-50.59 19.49-59.28 48.56a3.13 3.13 0 01-.81.65c-56.38 23.21-83.78 67.74-83.78 136.14 0 75.36-13.29 91.42-35.31 118-1.6 1.93-3.23 3.89-4.89 5.93a35.16 35.16 0 00-4.65 37.62c6.17 13 19.32 21.07 34.33 21.07H410.5c14.94 0 28-8.06 34.19-21a35.17 35.17 0 00-4.61-37.66zM256 480a80.06 80.06 0 0070.44-42.13 4 4 0 00-3.54-5.87H189.12a4 4 0 00-3.55 5.87A80.06 80.06 0 00256 480z"
          />
        </svg>
        <div
          className="flex"
        />
      </div>
      <div
        onClick={[Function]}
        style={
          Object {
            "cursor": "pointer",
          }
        }
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 16 16"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M7 14s-1 0-1-1 1-4 5-4 5 3 5 4-1 1-1 1H7zm4-6a3 3 0 1 0 0-6 3 3 0 0 0 0 6z"
          />
          <path
            d="M5.216 14A2.238 2.238 0 0 1 5 13c0-1.355.68-2.75 1.936-3.72A6.325 6.325 0 0 0 5 9c-4 0-5 3-5 4s1 1 1 1h4.216z"
            fillRule="evenodd"
          />
          <path
            d="M4.5 8a2.5 2.5 0 1 0 0-5 2.5 2.5 0 0 0 0 5z"
          />
        </svg>
      </div>
    </div>
  </div>
  <div
    className="flex"
    style={
      Object {
        "width": "95%",
      }
    }
  >
    <div
      className="search-bar"
    >
      <svg
        fill="currentColor"
        height="1em"
        onClick={[Function]}
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": "black",
            "cursor": "pointer",
          }
        }
        viewBox="0 0 16 16"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M11.742 10.344a6.5 6.5 0 1 0-1.397 1.398h-.001c.03.04.062.078.098.115l3.85 3.85a1 1 0 0 0 1.415-1.414l-3.85-3.85a1.007 1.007 0 0 0-.115-.1zM12 6.5a5.5 5.5 0 1 1-11 0 5.5 5.5 0 0 1 11 0z"
        />
      </svg>
      <input
        className="css-l00huz"
        onChange={[Function]}
        placeholder="Search for chats"
        type="search"
        value=""
      />
    </div>
  </div>
</form>
`;
